### PR TITLE
REASSIGN OWNED

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/10862.

### CODE REVIEW

**Comment by ericharmeling** [comment]
@fabiog1901 FYI

**Review by RichardJCai — CHANGES_REQUESTED** [review]
Mostly looks good to me, there's a missing privilege check on this that I'll make a PR shortly to fix.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11014#-:-MgNKh5X0xI8QtJk1Krs:b7qah3v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

___
*[v21.1/reassign-owned.md, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNK4Pi68Lcid-69dNm:-MgNK4Pi68Lcid-69dNn:bwo0vfi) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L15)):*
> ```Markdown
> ## Required privileges
> 
> - To change the owner of objects in a database owned by a specific role or user, the user executing the command must be a member of the [`admin` role](authorization.html#admin-role).
> ```

The user executing the command doesn't have to be a member of the admin role but must be a member of the `old` and `new` role.

See: 
https://www.postgresql.org/docs/current/sql-reassign-owned.html

Right now we actually don't have any privilege checks which is a bug I'm gonna make a PR right now to fix it.
___
*[v21.1/reassign-owned.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNKLGV88gYqSoq6yig:-MgNKLGV88gYqSoq6yih:b-zfhrsi) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L16)):*
> ```Markdown
> 
> - To change the owner of objects in a database owned by a specific role or user, the user executing the command must be a member of the [`admin` role](authorization.html#admin-role).
> - The target role (i.e., the desired role of the objects) must have the `CREATE` [privilege](authorization.html#assign-privileges) on the current database.
> ```

Does not need create either
___
*[v21.1/reassign-owned.md, line 87 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNKZ3A-NfFIy8gbb8Y:-MgNKZ3A-NfFIy8gbb8Z:bqadjli) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L87)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ sql
> > REASSIGN OWNED BY cockroachlabs TO movrlabs;
> ```

You'd actually have to GRANT movrlabs to cockroachlabs first  for this to succeed but the privilege check is bugged right now.

**Review by RichardJCai — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11014)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

___
*[v21.1/reassign-owned.md, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNK4Pi68Lcid-69dNm:-MgNSZH50OSxdynluyy3:byjc6c0) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L15)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

The user executing the command doesn't have to be a member of the admin role but must be a member of the `old` and `new` role.

See: 
https://www.postgresql.org/docs/current/sql-reassign-owned.html

Right now we actually don't have any privilege checks which is a bug I'm gonna make a PR right now to fix it.
</blockquote></details>

Actually sorry, since this is for 21.1 and we wouldn't be backporting my fix, just change it to 
the user executing it must be either a member of the admin role or must be a member of the new role and have CREATE. It'll be changed in the next version

**Review by ericharmeling — COMMENTED** [review]
TFTR @richardjcai 

I followed the privileges suggestion in your last comment, but held off on all other suggestions. PTAL. 

We can always update if the 21.2 fix gets backported in a later minor release.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11014#-:-MgRFIfx10aIwK6ufOs8:b3ipmuv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai)

___
*[v21.1/reassign-owned.md, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNK4Pi68Lcid-69dNm:-MgRFm3OBqfU6F9Abcnv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L15)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Actually sorry, since this is for 21.1 and we wouldn't be backporting my fix, just change it to 
the user executing it must be either a member of the admin role or must be a member of the new role and have CREATE. It'll be changed in the next version
</blockquote></details>

Done.
___
*[v21.1/reassign-owned.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/11014#-MgNKLGV88gYqSoq6yig:-MgRF_3_6uAYk_pm-n47:bft11ai) ([raw file](https://github.com/cockroachdb/docs/blob/f6479e7efc7134e0a3b70393f86f1eab25e48009/v21.1/reassign-owned.md#L16)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Does not need create either
</blockquote></details>

I get the following in 21.1.6:

~~~
demo@127.0.0.1:26257/movr> REASSIGN OWNED BY cockroachlabs TO movrlabs;
ERROR: user movrlabs does not have CREATE privilege on database movr
~~~

But maybe this is part of the bug you are talking about?

**Comment by RichardJCai** [comment]
> Previously, RichardJCai (Richard Cai) wrote…
> I get the following in 21.1.6:
> 
> ```
> demo@127.0.0.1:26257/movr> REASSIGN OWNED BY cockroachlabs TO movrlabs;
> ERROR: user movrlabs does not have CREATE privilege on database movr
> ```
> 
> But maybe this is part of the bug you are talking about?

Yeah I realized the privilege check is "incorrect" filed a separate issue about it. This LGTM for now

**Review by RichardJCai — APPROVED** [review]
Thanks for documenting this!

File: v21.1/reassign-owned.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+## Example
+
+{% include {{page.version.version}}/sql/movr-statements.md %}
+
+### Change the owner of all tables in a database
+
+Suppose that the current owner of the `users`, `vehicles`, and `rides` tables in the `movr` database is a role named `cockroachlabs`.
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:39]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> CREATE ROLE cockroachlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:44]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> GRANT CREATE ON DATABASE movr TO cockroachlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:49]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> ALTER TABLE users OWNER TO cockroachlabs;
+> ALTER TABLE vehicles OWNER TO cockroachlabs;
+> ALTER TABLE rides OWNER TO cockroachlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:56]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  public      | user_promo_codes           | table | demo          |                   0 | NULL
+  public      | users                      | table | cockroachlabs |                  50 | NULL
+  public      | vehicle_location_histories | table | demo          |                1000 | NULL
+  public      | vehicles                   | table | cockroachlabs |                  15 | NULL
+(6 rows)
+~~~
+
+Now suppose you want to change the owner for all of the tables owned by `cockroachlabs` to a new role named `movrlabs`.
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:75]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> CREATE ROLE movrlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:80]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> GRANT CREATE ON DATABASE movr TO movrlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:85]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/reassign-owned.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+~~~ sql
+> REASSIGN OWNED BY cockroachlabs TO movrlabs;
+~~~
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/reassign-owned.md:90]:
```suggestion
{% include_cached copy-clipboard.html %}
```

**Review by ianjevans — APPROVED** [review]
Just some include_cached nits, looks good.

**Comment by ericharmeling** [comment]
TFTR!

These `include_cached`s keep getting me 😆

**Pull request merged**